source/linux/Foundational_Components/U-Boot/Apps-SPL-Debug-OpenOCD.rst

```diff
@@ -1,12 +1,515 @@
+            .
+            .
+
+         - These dead loops are in assembly and are specific to the R5 or A53 processor in
+           the AM62x, based on which :file:`start.S` file you opened and modified above.
+
+   .. note::
+
+      The location of a debug dead loop on all K3 devices will vary based on the section of the SPL we are trying to debug.
+
+   .. ifconfig:: CONFIG_part_family in ('AM335X_family', 'AM437X_family', 'AM57X_family')
+
+      .. note::
+
+         On AM335x/AM437x/AM57xx, a debug dead loop could be added in the *early_system_init* function at
+         the file path :file:`ti-u-boot/arch/arm/mach-omap2/hwinit-common.c`. The location of a debug dead loop
+         on all platforms will vary based on the section of the SPL we are trying to debug.
+
+#. Move the modified u-boot binaries to the SD card:
+
+   After building u-boot, on AM62x, copy/replace the :file:`tiboot3.bin`, :file:`tispl.bin`, and :file:`u-boot.img`
+   u-boot binaries to the SD card's boot partition and insert the SD card in the board.
+
+   .. ifconfig:: CONFIG_part_family in ('AM335X_family', 'AM437X_family', 'AM57X_family')
+
+      .. note::
+
+         On AM335x/AM437x/AM57xx, copy :file:`MLO` and :file:`u-boot.img` to the SD card.
+
+   .. ifconfig:: CONFIG_part_family in ('AM65X_family')
+
+      .. note::
+
+         On AM65x, copy
+         :file:`sysfw.itb`, :file:`tiboot3.bin`, :file:`tispl.bin`, and :file:`u-boot.img` to the SD card.
+
+#. Power on the SK-AM62B-P1 board:
+
+   With a micro-USB cable, plug in the board to the host machine via the JTAG
+   XDS110 port:
+
+   .. Image:: /images/sk-am62b-p1-top-jtag.png
+      :width: 625
+
+   With a USB-C cable, plug in the board to power using either USB-C port:
+
+   .. Image:: /images/sk-am62b-p1-top-power.png
+      :width: 625
+
+"""""""""""""""""""
+Using OpenOCD & GDB
+"""""""""""""""""""
+
+OpenOCD and GDB are useful command-line tools used for debugging on embedded systems. Here,
+OpenOCD communicates with JTAG, and GDB communicates with OpenOCD.
+
+#. `Download, install, and setup OpenOCD and GDB-Multiarch <https://docs.u-boot.org/en/latest/board/ti/k3.html#step-1-download-and-install-openocd>`_.
+
+#. Connect to the SK-AM62B-P1 board using OpenOCD (board connected via JTAG XDS110 and powered on):
+
+   .. code-block:: console
+
+      $ openocd -f ~/openocd/tcl/board/ti_am625evm.cfg
+      Open On-Chip Debugger 0.12.0+dev-01925-g169d463a3 (2025-04-04-11:08)
+      Licensed under GNU GPL v2
+      .
+      .
+      Info : Listening on port 6666 for tcl connections
+      Info : Listening on port 4444 for telnet connections
+      Info : XDS110: connected
+      .
+      .
+      Info : XDS110: connected to target via JTAG
+      .
+      .
+      Info : JTAG tap: am625.cpu tap/device found: 0x0bb7e02f (mfg: 0x017 (Texas Instruments), part: 0xbb7e, ver: 0x0)
+      Info : [am625.axi_ap] Examination succeed
+      Info : [am625.cpu.sysctrl] starting gdb server on 3333
```

Comment:
So another thing we may want to consider mentioning is the part's security variant. Most newer SoCs will block any access to the sysctrl core and SE parts will absolutely deny any JTAG access unless we add the debug entry into the tiboot3 cert or modify some pins. 

We don't have to be that descriptive I don't think, just something along the of 'the security variant of the part will determine which core you have access to via the JTAG interface' kind of notice

Reply:
It is good to be descriptive, but I wonder if that could be added to upstream uboot documentation.